improve smoke test docs (#4421)
Kong/insomnia: packages/insomnia-smoke-test/README.md

## Structure

| Folder       | Purpose                           |
| ------------ | --------------------------------- |
| `/cli`       | tests for inso                    |
| `/tests`     | tests for Insomnia                |
| `/playwright`| test helpers                      |
| `/server`    | Express server used by the tests  |
| `/fixtures`  | data used by tests and the server |

## Run Insomnia app smoke tests

### Development method

In one terminal run the watcher

```shell

> npm run app-start-playwright    # Run watcher

Reviewer: Watcher would no longer be `app-start-playwright` after this PR, but `watch:app`
```suggestion
npm run watch:app   # Run watcher
```

```

In a second terminal run/debug/step through smoke tests

```shell
# Run tests
npm run test:smoke:dev
# Debug tests with verbose output
DEBUG=pw:browser,pw:api npm run test:smoke:dev
# Step through tests with playwright inspector

> PWDEBUG=1 npm run test:smoke:dev

Reviewer 2: the way these are presented, it's always confusing at first when I look at them because it's like it's suggesting to me to run them all at once.  if they were split into separate codeblocks it might read easier and would allow you to comment outside of the script

Reviewer 2: Run tests

```shell
npm run test:smoke:dev
```

Debug tests with verbose output

```shell
DEBUG=pw:browser,pw:api npm run test:smoke:dev
```

Step through tests with playwright inspector

```shell
PWDEBUG=1 npm run test:smoke:dev
```

Author: I started with that but saw inconsistencies elsewhere but I don't mind changing it back

Reviewer 2: whatever you want to do is cool with me.

```

### Build and package methods

> Build = js bundle to load into an electron client
> Package = executable binary `.dmg` or `.exe`

Reviewer: ```suggestion
It's possible to run the smoke tests for:
- A `build`, the JS bundle that is loaded into an electron client;
- A `package`, the executable binary (e.g. `.dmg` or `.exe`);
```

> 

Reviewer: ```suggestion
For `build`:
```

```shell
npm run app-build:smoke             # Transpile js bundle

> npm run test:smoke:package          # Run tests

Reviewer: There's a typo here, you probably mean `build` instead of `package`:
```suggestion
npm run test:smoke:build          # Run tests
```

```

> ```shell

Reviewer: ```suggestion
For `package`:
```shell
```

## Structure

| Folder       | Purpose                           |
| ------------ | --------------------------------- |
| `/cli`       | tests for inso                    |
| `/tests`     | tests for Insomnia                |
| `/playwright`| test helpers                      |
| `/server`    | Express server used by the tests  |
| `/fixtures`  | data used by tests and the server |

## Run Insomnia app smoke tests

### Development method

In one terminal run the watcher

```shell
npm run watch:app   # Run watcher
```

In a second terminal run/debug/step through smoke tests

```shell
# Run tests
npm run test:smoke:dev
# Debug tests with verbose output
DEBUG=pw:browser,pw:api npm run test:smoke:dev
# Step through tests with playwright inspector
PWDEBUG=1 npm run test:smoke:dev
```

### Build and package methods

It's possible to run the smoke tests for:

> - A `build`, the JS bundle that is loaded into an electron client;

Reviewer: ```suggestion

- A `build`, the JS bundle that is loaded into an electron client;
```

- A `package`, the executable binary (e.g. `.dmg` or `.exe`);

> For `build`:

Reviewer: ```suggestion

For `build`:

```

```shell
npm run app-build:smoke             # Transpile js bundle
npm run test:smoke:build          # Run tests
```

> For `package`:

Reviewer: ```suggestion
For `package`:

```

> It's possible to run the smoke tests for:
> - A `build`, the JS bundle that is loaded into an electron client;
> - A `package`, the executable binary (e.g. `.dmg` or `.exe`);
> For `build`:
> ```shell
> npm run app-build:smoke             # Transpile js bundle
> npm run test:smoke:build          # Run tests
> ```
> 
> For `package`:
> ```shell

Reviewer: @jackkav  this should fix the lint errors:
```suggestion
It's possible to run the smoke tests for:

- A `build`, the JS bundle that is loaded into an electron client;
- A `package`, the executable binary (e.g. `.dmg` or `.exe`);

For `build`:

```shell
npm run app-build:smoke         # Transpile js bundle
npm run test:smoke:build          # Run tests
```

For `package`:

```shell
```
